File: 240704/3n + 1 수열/3n-plus-1-sequence.py
n = int(input())
cnt = 0

while True:
    #디버깅 해보면 break문이 먼저 들어가야 함.
    if n == 1 :
        break

    if n % 2 == 0 :
        n = n // 2
        cnt+=1
    else:
        n = (n*3)+1
        cnt+=1
    

print(cnt)

File: 241125/3N + 1 수열/3n-plus-1-sequence.c
#include <stdio.h>

int main() {
    // 여기에 코드를 작성해주세요.
    int n, cnt=0;
    scanf("%d", &n);
    
    while (1){
        if(n==1){
            break;
        }
        if(n%2==0){
            n=n/2;
            cnt++;
            continue;
        }
        if(n%2==1){
           n=n*3+1;
           cnt++;
           continue;
        }
    }
    printf("%d", cnt);
    return 0;
}